### PR TITLE
pod remotes

### PR DESCRIPTION
**Pull request by giograno**
Some initial documentation for the Cloud Pods remotes (cc @cabeaulac)

### CODE REVIEW

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+weight: 5
+categories: ["LocalStack Pro", "Tools", "Persistence", "CLI"]
+description: >
+  Guide to the usage of Cloud Pods remotes
+aliases:
+  - /tools/cloud-pods/remotes/
+---
+
+{{< alert title="Beta">}}
+Cloud Pod Remotes is a feature that is currently in beta.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:12]:
```suggestion
Cloud Pod Remotes is a feature that is currently in beta (available in the Team product tier).
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+{{< /alert >}}
+
+A remote is the place where your Cloud Pods are stored.
+By default, we store the Cloud Pods' artifacts in the LocalStack platform.
+However, for various reasons, e.g., data regulation and sovereignty, you might want to store Cloud Pods in a different location that is under your control.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:17]:
```suggestion
Although we provide the highest levels of security and isolation (incl. encryption in transit and at rest), there may be situations (e.g., due to data regulation and sovereignty) where your organization does not permit storing Cloud Pod assets in a remote storage infrastructure. 
Hence, you may want to persist Cloud Pods in a different storage location that is on-premise and fully under your control.
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+We currently offer two alternative remotes:
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:19]:
```suggestion
We currently offer two alternative types of remotes:
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+- S3 bucket remote storage;
+- [ORAS](https://oras.land/) (OCI Registry as Storage) remote storage.
+
+Our CLI has been expanded and now offers commands to create and delete remotes:
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:23]:
```suggestion
Our Cloud Pods command-line interface (CLI) has been expanded and now offers commands to create and delete remotes:
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+```bash
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:25]: nit: we could use the `<command>` markup macro here.. 👍

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+Options:
+  -h, --help  Show this message and exit.
+
+Commands:
+  add     Add a remote
+  delete  Delete a remote
+```
+
+Moreover, the `localstack pod remotes` command will show all the registered remotes.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:39]: nit: we briefly discussed this in the past - should be good to merge as-is for now, 👍  but maybe we can unify this a bit over time (at least we could add a `pod remote list` command, and see how the `remotes` command shortcut compares to other parts of the CLI)

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+## S3 bucket remote storage
+The S3 storage lets you store Cloud Pods into any arbitrary S3 bucket.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:42]:
```suggestion
The S3 remote lets you store Cloud Pods assets into an existing S3 bucket in a real AWS account.
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+The first step is to export proper AWS credentials in the terminal session.
+
+```bash
+export AWS_ACCESS_KEY_ID=...
+export AWS_SECRET_ACCESS_KEY=...
+```
+
+A possible option is to obtain credentials via [AWS SSO CLI](https://github.com/synfinatic/aws-sso-cli).
+
+Afterwards, we have to add a new remote that explicitly targets a S3 bucket.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:52]:
```suggestion
Afterwards, we add a new remote that explicitly targets an S3 bucket.
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+With the command below, we are creating a new remote called `s3-storage-aws` that will store the Cloud Pods' artifacts into a S3 bucket name `ls-pods-bucket-test`.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:53]:
```suggestion
With the command below, we are creating a new remote called `s3-storage-aws` that will store the Cloud Pods' artifacts into an S3 bucket named `ls-pods-bucket-test`.
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+The `access_key_id` and `secret_access_key` placeholder will make sure that the AWS credentials are correctly passed to the container.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:54]:
```suggestion
The `access_key_id` and `secret_access_key` placeholders will make sure that the AWS credentials are correctly passed to the container.
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+```bash
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:56]: nit: same as above, better to be replaced with a `<command>` (there's plenty of examples in this repo.. 👍 )

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+```
+
+With the `pod load` command we can later load the same pod saved into this remote:
+
+```bash
+localstack pod remote load s3-storage-aws
+```
+
+## ORAS remote storage
+The ORAS remote allows users to save Cloud Pods in registries such as Docker Hub, Nexus, or ECS registries.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:81]:
```suggestion
The ORAS remote allows users to save Cloud Pods in OCI-compatible registries such as Docker Hub, Nexus, or ECS registries.
ORAS is an acronym for _OCI Registry as Service_ - more details about this standard can be found [here](https://oras.land/).
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 13]
```diff
@@ -0,0 +1,116 @@
...
+export ORAS_PASSWORD=ILoveLocalStack1!
+```
+
+Then, I can use the CLI to create a new remote called `oras-remote`.
+
+```shell
+localstack pod remote add oras-remote oras://{oras_username}:{oras_password}@registry.hub.docker.com/<docker_hub_id>
+```
+
+Finally, I can save a pod with the just-configured remote, where `my-pod` is the name of the Cloud Pod while `oras-remote` is the name of the remote itself.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:99]:
```suggestion
Finally, we can save a pod with the just-configured remote, where `my-pod` is the name of the Cloud Pod while `oras-remote` is the name of the remote itself.
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+
+```shell
+localstack pod save my-pod oras-remote
+```
+
+Similarly, I could perform the opposite operation and load a Cloud Pod from `oras-remote` with the following command:
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:105]:
```suggestion
Similarly, we can perform the reverse operation and load a Cloud Pod from `oras-remote` with the following command:
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+
+```shell
+localstack pod load my-pod oras-remote
+```
+
+### Miscellaneous
+When not differently specified, all Cloud Pods' commands target our Platform as default remote.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:112]:
```suggestion
If not explicitly specified, all Cloud Pods commands target the LocalStack Platform as storage remote by default. 
Please note that the CLI needs to be properly authenticated against our Platform, and a Team/Enterprise subscription is required.
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+
+The configuration for a custom remote is saved inside the LocalStack container.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:114]:
```suggestion
The configurations for custom remotes are saved inside the [LocalStack volume directory](https://docs.localstack.cloud/references/filesystem/#localstack-volume-directory) and maintained by the LocalStack container itself.
```

File: content/en/user-guide/tools/cloud-pods/remotes/index.md [hunk 17]
```diff
@@ -0,0 +1,116 @@
...
+Therefore, if you want to share Cloud Pods within your team using a custom remote, each team member would need to have the same remote configuration.
```
Comment (whummer) [inline content/en/user-guide/tools/cloud-pods/remotes/index.md:115]:
```suggestion
Therefore, if you want to share Cloud Pods within your team using a custom remote, each team member would need to define the same remote configuration.
```

**Review by whummer — APPROVED** [review]
Nice docs, thanks for tackling this @giograno ! 🎉  Overall LGTM, just a few small nit suggestions regarding wording in a few places.. 👍

File: content/en/user-guide/tools/cloud-pods/remotes/index.md (hunk 6 above)
Comment (giograno) [reply content/en/user-guide/tools/cloud-pods/remotes/index.md:40]: 100%. Opened a [small PR](https://github.com/localstack/localstack-ext/pull/2138). We could take the change to introduce it into 2.3. What do you say?

**Pull request merged**